src/main/java/frc/robot/subsystems/swerve/module/Module.java

```diff
@@ -0,0 +1,273 @@
+	) {
+		this.constants = constants;
+
+		this.encoder = encoder;
+		this.encoderSignals = encoderSignals;
+
+		this.steer = steer;
+		this.steerRequests = steerRequests;
+		this.steerSignals = steerSignals;
+
+		this.drive = drive;
+		this.driveRequests = driveRequests;
+		this.driveSignals = driveSignals;
+
+		this.targetState = new SwerveModuleState();
+		this.startingSteerPosition = new Rotation2d();
+		this.isClosedLoop = ModuleConstants.DEFAULT_IS_CLOSE_LOOP;
+		this.moduleInputs = new ModuleInputsAutoLogged();
+		this.driveInputs = new DriveInputsAutoLogged();
+		this.driveCouplingInputs = new DriveCouplingInputsAutoLogged();
+
+		updateInputs();
+		resetByEncoder();
+	}
+
+	public SysIdCalibrator.SysIdConfigInfo getSteerSysIdConfigInfo() {
+		return steer.getSysidConfigInfo();
+	}
+
+	public SysIdCalibrator.SysIdConfigInfo getDriveSysIdConfigInfo() {
+		return drive.getSysidConfigInfo();
+	}
+
+	public double toDriveMeters(Rotation2d angle) {
+		return Conversions.angleToDistance(angle, constants.wheelDiameterMeters());
+	}
+
+
+	private void fixDriveInputsCoupling() {
+		driveCouplingInputs.uncoupledVelocityAnglesPerSecond = ModuleUtils
+			.uncoupleDriveAngle(driveSignals.velocity().getLatestValue(), steerSignals.velocity().getLatestValue(), constants.couplingRatio());
+
+		driveCouplingInputs.uncoupledPositions = new Rotation2d[driveSignals.position().asArray().length];
+		for (int i = 0; i < driveCouplingInputs.uncoupledPositions.length; i++) {
+			Rotation2d steerDelta = Rotation2d
+				.fromRotations(steerSignals.position().asArray()[i].getRotations() - startingSteerPosition.getRotations());
+			driveCouplingInputs.uncoupledPositions[i] = ModuleUtils
+				.uncoupleDriveAngle(driveSignals.position().asArray()[i], steerDelta, constants.couplingRatio());
+		}
+	}
+
+	public void updateInputs() {
+		steer.updateSimulation();
+		drive.updateSimulation();
+
+		encoder.updateInputs(encoderSignals.position());
+		steer.updateInputs(steerSignals.position(), steerSignals.velocity(), steerSignals.current(), steerSignals.voltage());
+		drive.updateInputs(driveSignals.position(), driveSignals.velocity(), driveSignals.current(), driveSignals.voltage());
+
+		fixDriveInputsCoupling();
+
+		driveInputs.velocityMetersPerSecond = toDriveMeters(driveCouplingInputs.uncoupledVelocityAnglesPerSecond);
+		driveInputs.positionsMeters = Arrays.stream(driveCouplingInputs.uncoupledPositions).mapToDouble(this::toDriveMeters).toArray();
+
+		moduleInputs.isClosedLoop = isClosedLoop;
+		moduleInputs.targetState = targetState;
+
+		Logger.processInputs(constants.logPath(), moduleInputs);
+		Logger.processInputs(constants.logPath() + "Drive", driveInputs);
+		Logger.processInputs(constants.logPath() + "Drive", driveCouplingInputs);
+	}
+
+
+	public void setClosedLoop(boolean closedLoop) {
+		isClosedLoop = closedLoop;
+	}
+
+	public void setBrake(boolean brake) {
+		steer.setBrake(brake);
+		drive.setBrake(brake);
+	}
+
+	public void resetByEncoder() {
```

Comment:
```suggestion
	public void resetSteerByEncoder() {
```


You can also reset the drive motor, even though it serves no purpose since in the module you just control its velocity.